mystery_word.py

```diff
@@ -0,0 +1,206 @@
+    return difficulty
+
+
+def choose_difficulty(mode):
+    """Allows user to enter difficulty and runs recursively on bad input"""
+    if mode == "e" or mode == "easy":
+        return "e"
+    if mode == "m" or mode == "medium":
+        return "m"
+    if mode == "h" or mode == "hard":
+        return "h"
+    if mode == "n" or mode == "nightmare":
+        return "n"
+    if mode == "q" or mode == "quit":
+        return "q"
+    else:
+        print("Try Again\n")
+        return choose_difficulty(input(">>").lower())
+
+
+def get_wordlist(file_address):
+    """opens file and reads it as one large string
+    builds a list of words by splitting the string on ' '. """
+    with open(file_address) as file:
+        file_string = file.read()
+    word_list = file_string.split()
+    return word_list
+
+
+def generate_word(file_address):
+    """receives the word list from get_wordlist()
+    chooses random word within the constraints of word_length
+    and is_common_noun()
+    If user has chosen to quit or move to nightmare mode, return 0"""
+    if difficulty == "q" or difficulty == "n":
+        return 0
+    wordlist = get_wordlist(file_address)
+    random_int = random.randint(0, len(wordlist)-1)
+    random_word = wordlist[random_int]
+    if is_common_noun(random_word) and is_proper_word_length(random_word):
+        print("Your mystery word "
+              "is {} characters long.".format(len(random_word)))
+        return random_word
+    else:
+        return generate_word(file_address)
+
+
+def is_common_noun(word):
+    """ Checks if word is a common noun.
+    Prevents redundancies such as Titanic and titanic. """
+    if word == word.lower():
+        return True
+    else:
+        return False
+
+
+def is_proper_word_length(word):
+    """ Checks word's length against the constraints of game's difficulty"""
+    if difficulty == "e":
+        return 4 <= len(word) <= 6
+    elif difficulty == "m":
+        return 6 <= len(word) <= 10
+    elif difficulty == "h":
+        return 10 <= len(word)
+    else:
+        return True
+
+
+def validate_input(user_input):
+    """Checks for bad input and returns the input unmodified or an error message
+    Errors: multiple characters, non-alphabetic input and redundant tries
+    Also allows user to quit with 'quit'. """
+    if user_input == 'quit':
+        return "Thanks for playing.\n"
+    elif len(user_input) > 1:
+        return "One letter at a time!\n"
+    elif user_input == "":
+        return "Give it another shot.\n"
+    elif not user_input.isalpha():
+        return "Letters only.\n"
+    elif user_input in guessed_letters:
+        return "You already tried that one."
+    else:
+        return user_input
+
+
+def check_guessed_letter_in_mystery_word(user_input):
+    """ Checks if the user's guess is in the mystery word.
+    Records guess in guessed_letters
+    Decrements guess if not. """
+    global discovered_letters
+    global guesses, guessed_letters
```

Comment:
HOT TEARS OF GLOBAL VARIABLES

I think this probably won't happen again since we now have objects to hold our state, but when using functions, try to pass in everything the function needs instead of using global scope.